libtlafmt/src/renderer/indent.rs

```diff
@@ -0,0 +1,392 @@
+use crate::token::Token;
+
+use super::Indent;
+
+/// Process the token buffer, reducing the indentation of excessively indented
+/// blocks relative to the previous line.
+///
+/// An excessively indented block is one in which all tokens within it are more
+/// than 1 level deeper in indentation than the parent. It rewrites this:
+///
+/// ```text
+///    LET n == Len(InitVals)
+///            gg == CHOOSE g:
+///                    \E e \in PosReal:
+///                        /\ \A r \in OpenInterval(a - e, b + e):
+///                                    D[X |-> 42] = 0
+/// ```
+///
+/// Into this:
+///
+/// ```text
+///    LET n == Len(InitVals)
+///        gg == CHOOSE g:
+///            \E e \in PosReal:
+///                /\ \A r \in OpenInterval(a - e, b + e):
+///                     D[X |-> 42] = 0
+/// ```
+///
+pub(super) fn limit_indents(buf: &mut [(Token<'_>, Indent)]) {
```

Comment:
I've known this rewrite pass was needed for a while, but it was a requirement in order to indent bounded quantifications in order to fix the spec breakage.